lib/unleash/metrics_reporter.rb

```diff
@@ -15,48 +14,35 @@ def initialize
     end
 
     def generate_report
-      now = Time.now
+      metrics = Unleash&.engine&.get_metrics()
```

Comment:
Why is this optional here? It doesn't seem optional in client.rb

Reply:
Good question! This may be my Rust opinions coming through here. The Client is owns the engine and makes it available to everything else when it's started up. Everything else may or may not have access to it, so this is paranoia and memory and ownership hygiene